src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3,16 +3,1430 @@
+        dialog.CloseButtonText(buttonText);
+
+        _showDialogHandlers(*this, dialog);
+    }
+
+    // Method Description:
+    // - Displays a dialog for errors found while loading or validating the
+    //   settings. Uses the resources under the provided  title and content keys
+    //   as the title and first content of the dialog, then also displays a
+    //   message for whatever exception was found while validating the settings.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens. See _ShowDialog for details
+    // Arguments:
+    // - titleKey: The key to use to lookup the title text from our resources.
+    // - contentKey: The key to use to lookup the content text from our resources.
+    void TerminalPage::ShowLoadErrorsDialog(const winrt::hstring& titleKey,
+                                            const winrt::hstring& contentKey,
+                                            HRESULT settingsLoadedResult)
+    {
+        auto title = _resourceLoader.GetLocalizedString(titleKey);
+        auto buttonText = _resourceLoader.GetLocalizedString(L"Ok");
+
+        Controls::TextBlock warningsTextBlock;
+        // Make sure you can copy-paste
+        warningsTextBlock.IsTextSelectionEnabled(true);
+        // Make sure the lines of text wrap
+        warningsTextBlock.TextWrapping(TextWrapping::Wrap);
+
+        winrt::Windows::UI::Xaml::Documents::Run errorRun;
+        const auto errorLabel = _resourceLoader.GetLocalizedString(contentKey);
+        errorRun.Text(errorLabel);
+        warningsTextBlock.Inlines().Append(errorRun);
+
+        if (FAILED(settingsLoadedResult))
+        {
+            if (!_settingsLoadExceptionText.empty())
+            {
+                warningsTextBlock.Inlines().Append(_BuildErrorRun(_settingsLoadExceptionText, Resources()));
+            }
+        }
+
+        // Add a note that we're using the default settings in this case.
+        winrt::Windows::UI::Xaml::Documents::Run usingDefaultsRun;
+        const auto usingDefaultsText = _resourceLoader.GetLocalizedString(L"UsingDefaultSettingsText");
+        usingDefaultsRun.Text(usingDefaultsText);
+        warningsTextBlock.Inlines().Append(usingDefaultsRun);
+
+        Controls::ContentDialog dialog;
+        dialog.Title(winrt::box_value(title));
+        dialog.Content(winrt::box_value(warningsTextBlock));
+        dialog.CloseButtonText(buttonText);
+
+        _showDialogHandlers(*this, dialog);
+    }
+
+    // Method Description:
+    // - Displays a dialog for warnings found while loading or validating the
+    //   settings. Displays messages for whatever warnings were found while
+    //   validating the settings.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens. See _ShowDialog for details
+    void TerminalPage::ShowLoadWarningsDialog()
```

Comment:
The more I think on it, the more I want these guys to stay in `App`. We'd probably need to move all the static functions that the top of this file back into `App`, we could move `_settingsLoadExceptionText` back into `App`, and we wouldn't need `TerminalPage::SetSettingsLoadExceptionText` any longer